### PR TITLE
feat: [Feature] Use photo asset id instead of email address for profile picture in comments

### LINKED ISSUE #5385: [Feature] Show user's photo in comment from uploaded asset
## Summary

We should show the user's photo in comment card based on the asset id that has been uploaded by the user. This will make sure we're not trying to a image that does not exist i.e. user did not upload yet. We should also resize the uploaded photo for profile pictures and properly use http cache headers so that same photo is not loaded every time.

## Motivation

We fetch the user's photo by their email address. If an user did not upload their photo yet, we still try to load their photo by email address. This creates a 404 request that can be avoided.
We use a very small photo as profile picture. But the image can be much larger than that. When users are on comment page and there are comments from many users, a lot of images will be fetched to show the profile image. Also there is no cache control header so browser will load the photos every time they are requested. To address this problem, we need to create a thumbnail from the uploaded photo and store that. Also we should add cache headers to prevent fetching same photo multiple times.

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
Added the following changes in this PR. They are:
- Set author photo id in comments when new comment is added. Also update the photo id in comments when user changes or deletes their photo
- Resize the photo when user uploads their profile picture
- Display the profile photo in the comment card based on the uploaded asset id. Currently it shows by email address
- Added cache control header so that browser caches the profile pictures

Fixes #5385 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Unit tests
- Tested on local machine

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feature/set-profile-asset-id-in-comment 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.78 **(-0.01)** | 36.51 **(0)** | 33.93 **(-0.01)** | 55.33 **(0)**
 :green_circle: | app/client/src/comments/CommentCard/CommentCard.tsx | 71.82 **(0.15)** | 59.06 **(-0.14)** | 42.86 **(0)** | 72.19 **(0.17)**
 :green_circle: | app/client/src/components/ads/Text.tsx | 100 **(0)** | 78.57 **(3.57)** | 100 **(0)** | 100 **(0)**
 :green_circle: | app/client/src/pages/common/ProfileImage.tsx | 90 **(0)** | 68.75 **(6.25)** | 71.43 **(0)** | 88.89 **(0)**
 :red_circle: | app/client/src/utils/helpers.tsx | 65.15 **(-1.52)** | 39.18 **(-3.09)** | 46.88 **(-3.12)** | 62.18 **(-1.28)**</details>

### CODE REVIEW

File: app/client/src/comments/CommentCard/CommentCard.tsx [hunk 1]
```diff
@@ -443,7 +446,7 @@ function CommentCard({
         <HeaderSection>
           <ProfileImage
             side={25}
-            source={`/api/${USER_PHOTO_URL}/${authorUsername}`}
+            source={profilePhotoUrl}
```
Comment (rishabhsaxena) [inline app/client/src/comments/CommentCard/CommentCard.tsx:449]: we could conditionally pass the `source` prop, if the `authorPhotoId` is not defined or not, that would avoid `404s`

File: app/client/src/comments/CommentCard/CommentCard.tsx [hunk 2]
```diff
@@ -443,7 +446,7 @@ function CommentCard({
         <HeaderSection>
           <ProfileImage
```
Comment (rishabhsaxena) [inline app/client/src/comments/CommentCard/CommentCard.tsx:447]: Think we should update this at other places we're using the user's profile image as well?

**Review by rishabhsaxena — DISMISSED** [review]

File: app/client/src/comments/CommentCard/CommentCard.tsx (hunk 2 above)
Comment (nayan-rafiq) [reply app/client/src/comments/CommentCard/CommentCard.tsx:447]: It used in mention list but the underlying members API does not return the profile photo along with list of users. So I didn't change that behavior. We may consider that in future.

File: app/client/src/comments/CommentCard/CommentCard.tsx (hunk 1 above)
Comment (nayan-rafiq) [reply app/client/src/comments/CommentCard/CommentCard.tsx:449]: Yeah, that would be better. I need some help on that.

File: app/client/src/comments/CommentCard/CommentCard.tsx (hunk 2 above)
Comment (rishabhsaxena) [reply app/client/src/comments/CommentCard/CommentCard.tsx:447]: got it, actually if it's there with the user data, we could update it where we're showing the user's profile image: for example org users on (`/applications`) and settings

File: app/client/src/comments/CommentCard/CommentCard.tsx (hunk 1 above)
Comment (rishabhsaxena) [reply app/client/src/comments/CommentCard/CommentCard.tsx:449]: My bad, it's correctly implemented as of now too

**Comment by nayan-rafiq** [comment]
@rishabhsaxena I've rebased from release to fix failing automated tests. Would you please have a look again?

**Review by rishabhsaxena — APPROVED** [review]
FE changes LGTM!
Also have created another issue for updating other places where we could use the photo id:
https://github.com/appsmithorg/appsmith/issues/7610

**Pull request merged**